fdbserver/ptxn/test/RealTLog.actor.cpp

```diff
@@ -125,30 +134,38 @@ ACTOR Future<Void> getTLogCreateActor(std::shared_ptr<TestDriverContext> pTestDr
 		}
 	}
 
+	// delete old disk queue files
+	deleteFile(diskQueueFilename + "0." + tLogOptions.diskQueueExtension);
+	deleteFile(diskQueueFilename + "1." + tLogOptions.diskQueueExtension);
+	deleteFile(kvStoreFilename + "0." + tLogOptions.kvStoreExtension);
+	deleteFile(kvStoreFilename + "1." + tLogOptions.kvStoreExtension);
 	return Void();
 }
 
 // sent commits through TLog interface.
-ACTOR Future<Void> TLogDriverContext::sendCommitMessages_impl(std::shared_ptr<TestDriverContext> pTestDriverContext,
-                                                              TLogDriverContext* pTLogDriverContext) {
-	bool tLogReady = wait(pTLogDriverContext->pTLogContext->TLogStarted.getFuture());
+ACTOR Future<Void> TLogDriverContext::sendCommitMessages_impl(TLogDriverContext* pTLogDriverContext,
+                                                              uint16_t processID) {
+	state std::shared_ptr<TLogContext> pTLogContext = pTLogDriverContext->pTLogContextList[processID];
+	bool tLogReady = wait(pTLogContext->TLogStarted.getFuture());
 	ASSERT_EQ(tLogReady, true);
 
 	state Version prev = 0;
 	state Version next = 1;
 	state int i = 0;
-
 	for (; i < pTLogDriverContext->numCommits; i++) {
 		Standalone<StringRef> key = StringRef(format("key %d", i));
 		Standalone<StringRef> val = StringRef(format("value %d", i));
 		MutationRef m(MutationRef::Type::SetValue, key, val);
-		Tag tag(pTLogDriverContext->tagLocality, pTLogDriverContext->pTLogContext->tagProcessID);
 
 		// build commit request
 		LogPushData toCommit(pTLogDriverContext->ls);
 		UID spanID = deterministicRandom()->randomUniqueID();
 		toCommit.addTransactionInfo(spanID);
-		vector<Tag> tags = { tag };
+		vector<Tag> tags;
+		for (uint32_t tagID = 0; tagID < pTLogDriverContext->numTagsPerServer; tagID++) {
```

Comment:
This means every commit will use all tags, and tlog received the commit will index all tags.

Maybe add a TODO here: we could randomize tags to mimic real workload that have skewed tags. I.e., if the tlog is a primary location for a tag, it gets all tag data; otherwise, it gets a random portion of the data.

Reply:
many variations are possible .. we can iterate on them over time
I can add a todo